src/components/navigation/Header.svelte

```diff
@@ -19,13 +24,80 @@
 	function toggleNavbar() {
 		isMobileMenuOpen = !isMobileMenuOpen;
 	}
+
+	function checkOverflow() {
+		if (!navContainer) return;
+
+		const navWidth = navContainer.clientWidth;
+		const logoWidth = navContainer.querySelector('.logo-container')?.clientWidth || 0;
+		const themeWidth = 48;
```

Comment:
I don't understand what `themeWidth` is or why it's set to 48, or where the value of `32` comes from on the next line. Ideally, these two values should be self-explanatory, but if not there should be a comment that explains where the 48 and 32 come from, and why they aren't set to a different value.

Reply:
The `themeWidth` is the value for our theme toggler switch (now defined as a constant - ), and `32` is the extra padding/margin we need to calculate, when we want to trigger our burger icon for small screens. I have made the following changes based on your suggestions:

- I have now defined THEME_SWITCHER_WIDTH as 56 (the width of toggler) and EXTRA_PADDING (32 as calculated for extra padding) as constants and added comments to explain their purpose.